wordpress-mobile/WordPress-Android
# Bump Sentry Gradle plugin

## Description

This PR updates Sentry Gradle plugin to newest stable version, adding new features and improving build time.

### Source context upload

Different issues, but you can see that new issues will show the source code as well.

| Before | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/5845095/4f38cad0-3abd-449a-a22f-7ae4a82467eb) |
| --- | --- | 
| After | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/5845095/7924b22a-857e-4cda-b5c9-f7dfbe4236fa)|

### Disable dependencies report

We can always check the release and see what dependencies were used in a specific build that Sentry issue is coming from. This saves us from additional tasks in the build tree and makes the Sentry payload for each issue smaller.

### Disable Sentry tasks in `debug`

We don't use Sentry in debug builds. This removes all Sentry tasks in this build type, making builds quicker.

## To Test:

Please build the release variant (`vanillaRelease`) of both WordPress and Jetpack flavors and verify if issues are sent correctly. You can use the following snippet.

<details><summary>PATCH</summary>

```PATCH
Subject: [PATCH] tests: add a test for features in development generation
---
Index: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt b/WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
--- a/WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt	(revision 806913d9fb807250cecd5b24b36001d55ea4c255)
+++ b/WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt	(date 1710772966823)
@@ -5,6 +5,7 @@
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.distinctUntilChanged
+import com.automattic.android.tracks.crashlogging.CrashLogging
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.flow.firstOrNull
 import org.wordpress.android.R
@@ -67,6 +68,7 @@
     private val bloggingPromptsStore: BloggingPromptsStore,
     @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
     private val shouldAskPrivacyConsent: ShouldAskPrivacyConsent,
+    private val crashLogging: CrashLogging,
 ) : ScopedViewModel(mainDispatcher) {
     private var isStarted = false
 
@@ -161,6 +163,7 @@
         launch { loadMainActions(site) }
 
         updateFeatureAnnouncements()
+        crashLogging.sendReport(Throwable("Test crash"))
     }
 
     @Suppress("LongMethod")
```
</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - Sending Sentry issues

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual tests

3. What automated tests I added (or what prevented me from doing so)

    - None

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
